[DDP][Compile] Test to Ensure torch.compile works w/static_graph=True

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114621

Resolves https://github.com/pytorch/pytorch/issues/93672. This was
actually fixed by https://github.com/pytorch/pytorch/pull/103487 but I didn't
realize that PR also fixes torch compile at the time.

Differential Revision: [D51596148](https://our.internmc.facebook.com/intern/diff/D51596148/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @kiukchung @d4l3k @lucasllc

rohan-varma committed:
[DDP][Compile] Ensure torch.compile works w/static_graph=True

Resolves https://github.com/pytorch/pytorch/issues/93672. This was
actually fixed by https://github.com/pytorch/pytorch/pull/103487 but I didn't
realize that PR also fixes torch compile at the time.

Differential Revision: [D51596148](https://our.internmc.facebook.com/intern/diff/D51596148/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[DDP][Compile] Test to Ensure torch.compile works w/static_graph=True"


Resolves https://github.com/pytorch/pytorch/issues/93672. This was
actually fixed by https://github.com/pytorch/pytorch/pull/103487 but I didn't
realize that PR also fixes torch compile at the time.

Differential Revision: [D51596148](https://our.internmc.facebook.com/intern/diff/D51596148/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab mrshenli zhaojuanmao kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

davidberard98 commented on torch/testing/_internal/distributed/distributed_test.py:
```diff
@@ -9566,6 +9566,38 @@ def abort(device):
+            ddp_static = torch.nn.parallel.DistributedDataParallel(
+                model_clone,
+                device_ids=[self.rank],
+                static_graph=True
```
I think it was for every model (but hf_bert is probably still a good idea?)

and thanks for adding the test!

rohan-varma committed:
Update on "[DDP][Compile] Test to Ensure torch.compile works w/static_graph=True"


Resolves https://github.com/pytorch/pytorch/issues/93672. This was
actually fixed by https://github.com/pytorch/pytorch/pull/103487 but I didn't
realize that PR also fixes torch compile at the time.

Differential Revision: [D51596148](https://our.internmc.facebook.com/intern/diff/D51596148/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab mrshenli zhaojuanmao kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

wconstab reviewed (approved):
lgtm, thanks @rohan-varma !

rohan-varma commented:
Inductor accuracy failure looks unrelated: https://github.com/pytorch/pytorch/actions/runs/7010447659/job/19072184242.

rohan-varma commented:
@pytorchbot merge -f "CI flaky"

rohan-varma committed:
Update on "[DDP][Compile] Test to Ensure torch.compile works w/static_graph=True"


Resolves https://github.com/pytorch/pytorch/issues/93672. This was
actually fixed by https://github.com/pytorch/pytorch/pull/103487 but I didn't
realize that PR also fixes torch compile at the time.

Differential Revision: [D51596148](https://our.internmc.facebook.com/intern/diff/D51596148/)

cc H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab mrshenli zhaojuanmao kiukchung d4l3k lucasllc XilunWu tianyu-l

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "CI flaky"
